org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/AnnotationProcessingCompilerToolTest.java

```diff
@@ -93,7 +93,7 @@ public void test_github844() throws IOException {
 			+ "	last round: false\n"
 			+ "Discovered processor service org.eclipse.jdt.apt.tests.external.annotations.batch.BatchGenProcessor\n"
 			+ "  supporting [org.eclipse.jdt.apt.tests.external.annotations.batch.BatchGen]\n"
-			+ "  in jar:file:"  + _extJar.getCanonicalPath() +"!/\n"
+			+ "  in jar:"  + Path.of(_extJar.getCanonicalPath()).toUri().toURL().toString() +"!/\n"
```

Comment:
@iloveeclipse Can you check if this produced the correct result on windows?

Reply:
Interestingly, after switching launch config to use windows-1252 from UTF-8 the test did not fail anymore.
??? Why character encoding affects new lines ???

Anyway, looks OK now.

Reply:
Thanks for checking!